### PR TITLE
Improvements: CodeSandbox, Prettier, problem description, etc.

### PR DESCRIPTION
**Pull request by dtinth**
Changes:

1. The project has been updated to work with CodeSandbox.io. This means it’s possible to view, modify, and test the app without having to download, install, and run locally.

2. The problem has been clarified a bit more:

   1. Note that there are many bugs in this app and the candidate is expected to find them and fix them.

   2. Note that any API call may fail or be delayed for various reasons.

   3. Note that the candidate should avoid over-engineering and under-engineering.

3. The evaluation condition has been relaxed:

   1. “Functional programming paradigm is preferable.” has been removed as it could lead to overusing FP concepts than necessary/appropriate.

   2. “Functions do only one thing.” could lead to over-extracting functions. It has been replaced with “Good software engineering principles (e.g. SOLID principles) are followed as appropriate.”

4. The code base has been formatted with Prettier. I did this because CodeSandbox will also format code using Prettier.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,13 +34,13 @@ You can go back and forth through the steps anytime, but the whole process is co
...
-- Bugs are fixed based on the specifications.
-- Functional programming paradigm is preferable.
-- Functions do only one thing.
-- Identifiers are named meaningfully.
+- As many bugs are fixed as possible. It should be hard for us to find a remaining bug.
+- Good software engineering principles (e.g. SOLID principles) are followed as appropriate.
+- Identifiers are named meaningfully and consistently.
 - High coupling code blocks are adjacent to each other.
-- Possible errors are handled and exposed to users beautifully.
+- Possible errors are handled and exposed to users beautifully. Note: Any API call may result in an error due to invalid input, network disconnection, or server error. Due to network conditions, sometimes an API call may take longer time than usual.
```
Comment (ThisIsManta) [inline README.md]: Good idea! We should simulate that in our Firebase Clould Function (`server` directory).

File: README.md [hunk 2]
```diff
@@ -34,13 +34,13 @@ You can go back and forth through the steps anytime, but the whole process is co
...
 - Good choice of data structure is used.
 - Consistent indentation and formatting are followed.
```
Comment (ThisIsManta) [inline README.md]: Since we're going to use Prettier, we shall remove this line.

**Comment by ThisIsManta** [comment]
I added you as a collaborator. Now you can create a new branch against this repo without forking.

File: README.md (hunk 2 above)
Comment (dtinth) [reply README.md]: Not sure if the applicant has Prettier installed though… Should we reword this to e.g. “Code is formatted according to our Prettier configuration”?

File: README.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ You can go back and forth through the steps anytime, but the whole process is co
 
 ### Instructions
 
-- You are instructed to refactor mainly JavaScript files inside `client/src` directory.
+- You are instructed to improve the code mainly in JavaScript files inside `client/src` directory. There are many bugs hidden in this app, try to fix as many bugs as you can find. Also, try to refactor the code so that it becomes easier to maintain. We expect you to show your best work with highest attention to detail.
```
Comment (ThisIsManta) [inline README.md]: I want to emphasis `We expect you to show your best work with highest attention to detail.`. Shall we make it bold or something?

File: README.md (hunk 1 above)
Comment (dtinth) [reply README.md]: Yes... maybe in later PR?

Comment (ThisIsManta) [reply README.md]: I did so in the later commit.
